pkg/granted/credential_process.go

```diff
@@ -55,7 +56,8 @@ var CredentialProcess = cli.Command{
 		secureSessionCredentialStorage := securestorage.NewSecureSessionCredentialStorage()
 		clio.Debugw("running credential process with config", "profile", profileName, "url", c.String("url"), "window", c.Duration("window"), "disableCredentialProcessCache", cfg.DisableCredentialProcessCache)
 
-		useCache := !cfg.DisableCredentialProcessCache
+		cliNoCache := c.Bool("no-cache")
+		useCache := !cfg.DisableCredentialProcessCache || !cliNoCache
```

Comment:
I think you need to fix this logic so that useCache is false if either the config is set or the flag is set.
```suggestion
		useCache := !(cfg.DisableCredentialProcessCache || cliNoCache)
```